src/index.js

```diff
@@ -228,9 +230,22 @@ function applyMacros({
   return result
 }
 
-// eslint-disable-next-line consistent-return
-function getConfig(macro, filename, source) {
-  if (macro.options.configName) {
+function getConfig(macro, filename, source, options) {
+  const {configName} = macro.options
+  if (configName) {
+    let callOptions, configOptions, configError, configPath
+
+    if (Object.prototype.hasOwnProperty.call(options, configName)) {
+      if (options[configName] && typeof options[configName] !== 'object') {
+        // eslint-disable-next-line no-console
+        console.error(
+          `The macro plugin options' ${configName} property was not an object or null.`,
+        )
+      } else {
+        callOptions = options[configName]
+      }
+    }
+
     try {
       // lazy-loading it here to avoid perf issues of loading it up front.
```

Comment:
So I appreciate the comment, but maybe phobia was the right word for what influenced this decision. First, if you must synchronously load `cosmiconfig` no matter what I don't see how you can gain anything useful by doing it later, unless of course it turns out that babel-plugin-macros is loaded but no macros are used (which hardly seems like the case to optimize for).

Second, in optimizing for that case, you've actually hurt performance for the common case significantly. If you read the cosmiconfig docs you'll see that its caches are internal to the `explorer` object. You create a new explorer every time you process a file, so you are also doing a full config search every time you process a file, which is far more costly in the common case than what you were trying to avoid.

It's easy to refactor the code, but it seems like it'll be a bit tricker to fix the tests.

Reply:
Actually I suppose we can have the best of both worlds if we lazy load and cache the explorer

Reply:
This is done now.